### PR TITLE
Draft for corporate proxy

### PR DESCRIPTION
**Pull request by LioRei**
it's a draft (and ugly solution) for being able to install package behind a corporate proxy.

Optimal solution will be to take VS code configuration to do this but.... I'm not a python dev and I really don't know how to do this

### CODE REVIEW

File: platformio/package/download.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ def __init__(self, url, dest_dir=None):
         self._request = requests.get(
             url,
             stream=True,
+            verify=False,
```
Comment (maxgerhardt) [inline platformio/package/download.py:35]: This weakens security for *everyone* not using a proxy, opening them up to man-in-the-middle attacks, supply chain poisoning etc. Can we only ocnditionally do this if another option / environment variable is enabled, such as "Disable strict SSL verifcation"? Same for the line above.

Comment (LioRei) [reply platformio/package/download.py:35]: Of course, like this it's a big security concern ;)

We could base this on the environment variable [NODE_TLS_REJECT_UNAUTHORIZED](https://nodejs.org/api/cli.html#node_tls_reject_unauthorizedvalue) ?

Comment (ivankravets) [reply platformio/package/download.py:35]: Yes, we will provide environment variable to disable TLS Certificates verification

**Comment by mcspr** [comment]
afaik 'requests' automatically picks up `*_PROXY` environment variable, so it is not necessary to inject ourselves?
https://requests.readthedocs.io/en/latest/user/advanced/#proxies
https://docs.python.org/3/library/urllib.request.html#urllib.request.getproxies (with some exceptions)
https://github.com/psf/requests/blob/v2.28.1/requests/utils.py (/proxies/)

**Comment by LioRei** [comment]
that's what the doc says but that's not what happens in real life... I tried again and I confirm that even with the env variable, the request does not take care of the proxy

**Comment by ivankravets** [comment]
Resolved in https://github.com/platformio/platformio-core/issues/4432

**Pull request closed without merging**